main.ts

```diff
@@ -185,6 +185,23 @@ let guards: string[] = await Promise.all(args.file.map(async (fileName: string)
 
 	   let checks = []; // Note: the order here matters
 
+
+	//    def.properties.
+	   
+	function foo(props: object){
```

Comment:
Could you define this function somewhere near the definition of `InterfaceProp`? Also for the `prop` argument's type it should be `InterfaceProp`.